managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerFactoryImpl.java

```diff
@@ -147,15 +151,34 @@ private synchronized void refreshStats() {
 
         mbean.refreshStats(period, TimeUnit.NANOSECONDS);
         ledgers.values().forEach(mlfuture -> {
-            ManagedLedgerImpl ml = mlfuture.getNow(null);
-            if (ml != null) {
-                ml.mbean.refreshStats(period, TimeUnit.NANOSECONDS);
+            if (mlfuture.isDone() && !mlfuture.isCompletedExceptionally()) {
+                ManagedLedgerImpl ml = mlfuture.getNow(null);
+                if (ml != null) {
+                    ml.mbean.refreshStats(period, TimeUnit.NANOSECONDS);
+                }
             }
         });
 
         lastStatTimestamp = now;
     }
 
+    private synchronized void cacheEviction() {
+        long now = System.nanoTime();
+        long period = now - lastCacheEvictionTimestamp;
+        long maxTimestamp = now - period;
```

Comment:
umm.. it seems like `maxTimestamp` is `lastCacheEvictionTimestamp` only? so, we want to evict all the entries which are older than `lastCacheEvictionTimestamp` right? then can we directly use `lastCacheEvictionTimestamp`?

Reply:
Fixed